tests/integration/test_task_events_mgr.py

```diff
@@ -170,3 +172,44 @@ async def test__always_insert_task_job(
             '1/broken/01': 'submit-failed',
             '1/broken2/01': 'submit-failed'
         }
+
+
+async def test__process_message_failed_with_retry(one, start, log_filter):
+    """Log job failure, even if a retry is scheduled.
+
+    See: https://github.com/cylc/cylc-flow/pull/6169
+
+    """
+
+    async with start(one) as LOG:
+        fail_once = one.pool.get_tasks()[0]
+        # Add retry timers:
+        one.task_job_mgr._set_retry_timers(
+            fail_once, {
+                'execution retry delays': [1],
+                'submission retry delays': [1]
+            })
```

Comment:
Less fragile if this was just set in the workflow config for the test?

Reply:
Does it really make any difference? - the aim was to avoid fiddling with `one`. Can do if you insist.

Reply:
Surely creating a workflow config with these retry delays set is not any more involved than fiddling the internal retry timers?

Reply:
Not going to argue. Will change.

Reply:
Went away and had a look at it - essentially this is more unit-testy than integration test-like: If you set these timers in the config you still need to run this function here to set the retry timers. So I think that I'll leave it.